subprojects/normalization-java/src/main/java/org/gradle/api/internal/changedetection/state/NormalizedContentInfoCollector.java

```diff
@@ -65,6 +60,14 @@ public NormalizedContentInfo collect(File file) {
         }
     }
 
+    public NormalizedContentInfo collect(File file) {
+        try {
+            return collect(new FileInputStream(file));
+        } catch (FileNotFoundException e) {
+            throw new UncheckedIOException(String.format("Failed to create MD5 hash for file '%s' as it does not exist.", file), e);
+        }
+    }
```

Comment:
The `FileInputStream` isn't closed, let's use `try (new FileInputStream(file)) { /* ... */ }` instead.

Reply:
Good catch.  This is fixed.

I'd prefer to use try-with-resources wherever possible, but the problem in this case is that we have to catch the IOException from `Closeable.close()` but we just want to ignore it.  With try-with-resources, we need to catch it like any other IOException coming from the try block which means we can end up without a return value.  With a `finally` block, we can explicitly close it, catch and swallow the exception and still return whatever the main try block was going to return.  We have the same problem in `DefaultFileHasher` and really anywhere we want to just ignore the error on `close()`.